gen/exhibitor_tls_bootstrap.py

```diff
@@ -0,0 +1,113 @@
+import json
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, List
+from urllib.parse import urlparse
+
+from gen.exceptions import ExhibitorTLSBootstrapError
+
+
+PACKAGE_NAME = 'dcos-bootstrap-ca'
+BINARY_PATH = '/genconf/bin'
+CA_PATH = '/genconf/ca'
+
+
+def _check(config: Dict[str, Any]) -> List[str]:
+    """
+    Current constraints are:
+        config.yaml:
+            - exhibitor_tls_enabled == True,
+            - master_discovery must be static
+            - DC/OS variant must be enterprise
+    """
+    checks = [
+        (lambda: config.get('exhibitor_tls_enabled', False) == 'true',
+         'Exhibitor security is disabled'),
+        (lambda: config['master_discovery'] == 'static',
+         'Only static master discovery is supported'),
+        (lambda: config['dcos_variant'] == 'enterprise',
+         'Exhibitor security is an enterprise feature'),
+    ]
+
+    reasons = []
+    for check in checks:
+        if not check[0]():
+            reasons.append(check[1])
+
+    return reasons
+
+
+def _find_package(packages_json: str) -> str:
+    packages = json.loads(packages_json)
+    for package in packages:
+        if package.startswith(PACKAGE_NAME):
+            return package
+    raise Exception('{} package is not present'.format(PACKAGE_NAME))
+
+
+def _extract_package(package_path: str) -> None:
+    """ Extracts the dcos-bootstrap-ca package from the local pkgpanda
+    repository """
+    Path(BINARY_PATH).mkdir(exist_ok=True)
+    with tempfile.TemporaryDirectory() as td:
+        subprocess.run(['tar', '-xJf', package_path, '-C', td])
+
+        shutil.move(
+            Path(td) / 'bin' / PACKAGE_NAME,
+            Path(BINARY_PATH) / PACKAGE_NAME)
+
+
+def _init_ca(alt_names: List[str]) -> None:
+    """ Initializes the CA (generates a private key and self signed
+    certificate CA extensions) if the resulting files do not already exist """
+    root_ca_paths = [
+        Path(CA_PATH) / 'root-cert.pem', Path(CA_PATH) / 'root-key.pem']
+    if not all(map(lambda p: p.exists(), root_ca_paths)):
+        Path(CA_PATH).mkdir(mode=0o700, exist_ok=True)
```

Comment:
If the directory already exists, this call will not set the permissions to 0700. Perhaps add a chmod to be extra-safe.